app/gui/src/integration_test.rs

```diff
@@ -0,0 +1,155 @@
+//! Enso testing fixtures
+//!
+//! Support structures for high-level testing crates that must connect to the Project Manager.
+//! This includes integration tests and batch-mode application profiling.
+
+use crate::prelude::*;
+use enso_web::traits::*;
+
+use crate::executor::web::EventLoopExecutor;
+use crate::initializer::setup_global_executor;
+use crate::Ide;
+use enso_frp::future::EventOutputExt;
+use enso_web::Closure;
+use enso_web::HtmlDivElement;
+use ensogl::application::test_utils::ApplicationExt;
+use std::pin::Pin;
+
+
+
+/// Reexports of commonly-used structures, methods and traits.
+pub mod prelude {
+    pub use super::Fixture;
+
+    pub use super::wait_a_frame;
+    pub use crate::prelude::*;
+    pub use enso_frp::future::EventOutputExt;
+}
+
+
+
+// =======================
+// === IntegrationTest ===
+// =======================
+
+/// A fixture for each IDE integration tests. During setup, the executor and [`Ide`] structure are
+/// initialized.
+#[allow(missing_docs)]
+#[derive(Debug)]
+pub struct Fixture {
+    pub executor: EventLoopExecutor,
+    pub ide:      Ide,
+    pub root_div: HtmlDivElement,
+}
```

Comment:
This looks like an important structure, but the docs does not really explain what are its resposnibilities. For example, why does it create a new div element? What it will contain? These things should be explained in docs.

Reply:
Documented Fixture. Also, I noticed `root_div` was being created with a hard-coded ID "root". This value must match the value `Initializer` passes to `Application::new`, so I made the value a field in `Startup` that `Initializer` and `Fixture` can both use.